core/src/main/java/feign/Response.java

```diff
@@ -44,35 +44,150 @@
   private final String reason;
   private final Map<String, Collection<String>> headers;
   private final Body body;
+  private final Request request;
 
+  @Deprecated
+  /**
+   * @deprecated  To be removed in Feign 10
+   */
   private Response(int status, String reason, Map<String, Collection<String>> headers, Body body) {
     checkState(status >= 200, "Invalid status code: %s", status);
     this.status = status;
     this.reason = reason; //nullable
     this.headers = Collections.unmodifiableMap(caseInsensitiveCopyOf(headers));
     this.body = body; //nullable
+    this.request = null; //nullable
+  }
+
+  private Response(int status, String reason, Map<String, Collection<String>> headers, Body body, Request request) {
```

Comment:
I'd make this accept a Builder as an arg, as it lowers the arity of the ctor. then below do things like this..

this.status = builder.status;